client/src/Features/MicrosoftLearn/MicrosoftLearnFilterComponentUtils.tsx

```diff
@@ -0,0 +1,151 @@
+        case FilterType.types:
+          return getTypesToDisplay(filterStore.displayFilter[props.type]);
+      }
+    };
+
+    const displayOptions = getDisplayOptions();
+    if (props.type === FilterType.products) {
+      return (
+        <>
+          <MicrosoftLearnFilterComponent
+            styles={themedClassNames(FilterComponentStyles)}
+            filterType={props.type}
+            filterName={props.name}
+            filterOption={displayOptions}
+            search={true}
+            mainItemClickHandler={event => {
+              let target = event?.target as HTMLInputElement;
+              let value = target.getAttribute('aria-describedby');
+              if (value) {
+                let subItems: string[] = [...catalog?.products.values()]
+                  .filter(product => product.parentId && product.parentId === value)
+                  .map(product => product.id);
+                target.checked
+                  ? filterStore.addFilter(props.type, [...subItems, value])
+                  : filterStore.removeFilter(props.type, [...subItems, value]);
+              }
+            }}
+            subItemClickHandler={event => {
+              let target = event?.target as HTMLInputElement;
+              let type = FilterType.products;
+              let value = target.getAttribute('aria-describedby');
+              if (target.checked && value) {
+                filterStore.addFilter(type, [value]);
+              } else if (!target.checked && value) {
+                filterStore.removeFilter(type, [value]);
+              }
+            }}
+          />
+        </>
+      );
+    } else {
+      return (
+        <>
+          <MicrosoftLearnFilterComponent
+            styles={themedClassNames(FilterComponentStyles)}
+            filterType={props.type}
+            filterName={props.name}
+            filterOption={displayOptions}
+            search={props.type === FilterType.roles ? true : false}
+            mainItemClickHandler={event => {
+              let target = event?.target as HTMLInputElement;
+              let value = target.getAttribute('aria-describedby');
+              if (target.checked && value) {
+                filterStore.addFilter(props.type, [value]);
+              } else if (!target.checked && value) {
+                filterStore.removeFilter(props.type, [value]);
+              }
+            }}
+          />
+        </>
+      );
+    }
+  });
+};
+
+const FilterComponentStyles = ({ theme }: IThemeOnlyProps): FilterComponentStyles => ({
+  root: [
+    {
+      marginBottom: theme.spacing.l1
+    }
+  ],
+  title: [
+    {
+      color: theme.palette.neutralPrimary,
+      fontWeight: FontWeights.semibold,
+      display: window.innerWidth > 768 ? 'block' : 'none'
+    }
+  ],
+  search: [
+    {
+      marginTop: `calc(${theme.spacing.s1}*2)`
+    }
+  ],
+  optionsList: [
+    {
+      width: '260px',
```

Comment:
please update these values in `calc` format wherever possible

Reply:
I have updated the margins and paddings to use `calc` and `theme.spacings`. Only the `widths` remain in a few places, which is fixed.